languages/zh-tw/thingtalk.genie

```diff
@@ -0,0 +1,390 @@
+            if (proj.args[0] === 'picture_url')
+                return null;
+            if (proj.table.isAggregation)
+                return null;
+            let outParams = Object.keys(proj.table.schema.out);
+            let stream;
+            if (outParams.length === 1 && $options.flags.turking)
+                return null;
+            if (outParams.length === 1)
+                stream = C.tableToStream(proj.table, null);
+            else
+                stream = C.tableToStream(proj.table, proj.args);
+            return stream;
+        };
+        ('假如' | '如果') proj:projection_Any '變得' '不一樣 了' => {
+            if (proj.args[0] === 'picture_url')
+                return null;
+            if (proj.table.isAggregation)
+                return null;
+            let outParams = Object.keys(proj.table.schema.out);
+            let stream;
+            if (outParams.length === 1 && $options.flags.turking)
+                return null;
+            if (outParams.length === 1)
+                stream = C.tableToStream(proj.table, null);
+            else
+                stream = C.tableToStream(proj.table, proj.args);
+            return stream;
+        };
+    }
+    '當' table:complete_table ('變' | '改變' | '不一樣') '了' ('而且' | '且') filter:edge_filter ('時' | '的 時候') => {
+        if (!table.schema.is_monitorable || !C.checkFilter(table, filter) || table.schema.is_list)
+            return null;
+        table = C.addFilter(table, filter, $options);
+        if (!table)
+            return null;
+        return C.tableToStream(table, null);
+    };
+    '當' table:complete_table ('變' | '改變' | '不一樣') '了' ('而且' | '且') filter:atom_filter ('時' | '的 時候') => {
+        if (!table.schema.is_monitorable || !C.checkFilter(table, filter))
+            return null;
+        if ($options.flags.turking && table.schema.is_list)
+            return null;
+        table = C.addFilter(table, filter, $options);
+        if (!table)
+            return null;
+        return C.tableToStream(table, null);
+    };
+    edge_stream;
+    timer;
+}
+
+// FIXME add more...
+
+forward_get_do_command = {
+    ('給我' | '拿' | '取得') table:if_filtered_table ('然後' | '接著') action:thingpedia_action if !complete => new Ast.Statement.Command(table, [action]);
+    table:complete_get_command '，' ('然後' | '接著') action:thingpedia_action if !complete => new Ast.Statement.Command(table, [action]);
+    table:complete_get_command ('然後' | '接著') action:thingpedia_action if !complete => new Ast.Statement.Command(table, [action]);
+    table:complete_get_command '，' action:thingpedia_action if !complete => new Ast.Statement.Command(table, [action]);
+}
+
+complete_forward_get_do_command = {
+    forward_get_do_command if complete;
+}
+
+backward_get_do_command = {
+    action:thingpedia_action '當' ('拿到' | '取得' | '得到') table:complete_get_command ('時' | '的 時候') if !complete => new Ast.Statement.Command(table, [action]);
```

Comment:
so far, I think you have been following what we have in English line-by-line. 
But Chinese is such a different language, we should have different constructs. 

for example, this `backward_get_do_command` does not make sense as far as I know. we don't say "发消息当下雨的时候", we only go forward, never backward. 
Of course, we will learn more by eyeballing through paraphrase after the synthetic sentences are generated. But I just want to make sure you guys keep this in mind, instead of copying whatever we have done in English.

Reply:
Thank you!
I knew that Chinese would be quire different in some cases, so I remove all `with_....`. But the backward case, sometimes I do talk like that, although not very often, but it did happen. For example, '記得拿走喔！你要回去的時候'
But if that doesn't make sense to you, I think we can remove it.
What do you think?

Reply:
Hmm, maybe. let's keep it then.

Reply:
Got it!
Let's see what we will need after some synthetic experiments.